### ISSUE TITLE
Satisfy Warning in Network Processing Time Comparison

### ISSUE THREAD

**Issue by AVividLight** [Matt-and-Gib/gleemail#60]
The compiler is warning us that the following results in comparison between signed and unsigned integers. All variables here should be unsigned, so perhaps the arithmetic operators are affecting something. Fix this!

```
while(nowMS() - processStartTime < MAX_PROCESSING_TIME + (MAX_PROCESSING_TIME - previousProcessElapsedTime)) {
```

**nowMS**: unsigned long
**processStartTime**: unsigned long
**<**
**MAX_PROCESSING_TIME**: unsigned short
**previousProcessElapsedTime**: unsigned short

```
   /home/runner/work/gleemail/gleemail/src/networking.cpp: In member function 'short unsigned int Networking::doTimeSensitiveProcess(short unsigned int, short unsigned int, bool (Networking::*)(bool (Networking::*)(Queue<Message>&, QueueNode<Message>*), Queue<Message>&), bool (Networking::*)(Queue<Message>&, QueueNode<Message>*), Queue<Message>&)':
  /home/runner/work/gleemail/gleemail/src/networking.cpp:574:35: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
    while(nowMS() - processStartTime < MAX_PROCESSING_TIME + (MAX_PROCESSING_TIME - previousProcessElapsedTime)) { //Compile-time warning: comparing signed and unsigned int.
          ~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by AVividLight** [comment]
Perhaps review **Conversions** subheading from https://en.cppreference.com/w/cpp/language/operator_arithmetic

**Comment by AVividLight** [comment]
From https://docs.microsoft.com/en-us/previous-versions/aetzh118(v=vs.140)

`C++ promotions are "value-preserving." That is, the value after the promotion is guaranteed to be the same as the value before the promotion. In value-preserving promotions, objects of shorter integral types (such as bit fields or objects of type char) are promoted to type int if int can represent the full range of the original type. If int cannot represent the full range of values, then the object is promoted to type unsigned int. Although this strategy is the same as that used by ANSI C, **value-preserving conversions do not preserve the "signedness" of the object.**`
[Emphasis mine]

Therefore, if there is a difference in "size" anywhere, promotions must be causing the unsigned-ness to be lost.

**Comment by AVividLight** [comment]
Yes, type promotion was the cause. Fixed in e832bcced3aafadf5e5b6ad64d7ba5c8b64d6ff3
